src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -797,6 +801,15 @@ private void GetEvaluationResultsAndContinue(ArrayBuilder<EvalResultDataItem> ro
                 return null;
             }
 
+
+            if (declaredType.IsByRef)
+            {
+                // This shouldn't be possible, but if we do get here, bail now to prevent
+                // the code in MemberExpansion.CreateExpansion() from crashing.
+                Debug.Assert(false, "GetTypeExpansion: ByRef type passed in");
+                return null;
```

Comment:
If we do get here, we'll silently ignore the issue in retail builds. Consider replacing entire ```if``` with ```Debug.Assert(!declaredType.IsByRef);```.